From 1eaa52321395b5f58cbaf6caa27f2bec6f2fe406 Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Tue, 5 Apr 2011 22:06:07 +0000
Subject: [PATCH] NET-402 IMAP, NNTP, POP3 and SMTP classes uses BufferedReader
 for control channel, which does not follow the standard. Changed reader to
 CRLFLineReader.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/net/trunk@1089264 13f79535-47bb-0310-9956-ffa450edef68
---
 src/changes/changes.xml                                    | 4 ++++
 src/main/java/org/apache/commons/net/imap/IMAPSClient.java | 3 ++-
 src/main/java/org/apache/commons/net/nntp/NNTP.java        | 3 ++-
 src/main/java/org/apache/commons/net/pop3/POP3.java        | 3 ++-
 src/main/java/org/apache/commons/net/smtp/SMTP.java        | 3 ++-
 5 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index 3a9af476c..6cc6ed76c 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -67,6 +67,10 @@ This release is binary-compatible with 2.2, but there are some minor changes to
   - ftp.FTPSClient ctors no longer throw NoSuchAlgorithmException
   All users are recommended to upgrade.
 ">
+            <action issue="NET-402" dev="sebb" type="fix">
+            IMAP, NNTP, POP3 and SMTP classes uses BufferedReader for control channel, which does not follow the standard.
+            Changed reader to CRLFLineReader.
+            </action>
             <action issue="NET-401" dev="sebb" type="fix">
             FTP class uses BufferedReader for control channel, which does not follow the standard.
             Changed reader to CRLFLineReader.
diff --git a/src/main/java/org/apache/commons/net/imap/IMAPSClient.java b/src/main/java/org/apache/commons/net/imap/IMAPSClient.java
index 3e9e0111a..8ca261fa5 100644
--- a/src/main/java/org/apache/commons/net/imap/IMAPSClient.java
+++ b/src/main/java/org/apache/commons/net/imap/IMAPSClient.java
@@ -31,6 +31,7 @@
 import javax.net.ssl.SSLSocketFactory;
 import javax.net.ssl.TrustManager;
 
+import org.apache.commons.net.io.CRLFLineReader;
 import org.apache.commons.net.util.SSLContextUtils;
 
 /**
@@ -198,7 +199,7 @@ private void performSSLNegotiation() throws IOException
         _input_ = socket.getInputStream();
         _output_ = socket.getOutputStream();
         _reader =
-          new BufferedReader(new InputStreamReader(_input_,
+          new CRLFLineReader(new InputStreamReader(_input_,
                                                    __DEFAULT_ENCODING));
         __writer =
           new BufferedWriter(new OutputStreamWriter(_output_,
diff --git a/src/main/java/org/apache/commons/net/nntp/NNTP.java b/src/main/java/org/apache/commons/net/nntp/NNTP.java
index d712612e8..8710040bf 100644
--- a/src/main/java/org/apache/commons/net/nntp/NNTP.java
+++ b/src/main/java/org/apache/commons/net/nntp/NNTP.java
@@ -26,6 +26,7 @@
 import org.apache.commons.net.MalformedServerReplyException;
 import org.apache.commons.net.ProtocolCommandSupport;
 import org.apache.commons.net.SocketClient;
+import org.apache.commons.net.io.CRLFLineReader;
 
 /***
  * The NNTP class is not meant to be used by itself and is provided
@@ -169,7 +170,7 @@ protected void _connectAction_() throws IOException
     {
         super._connectAction_();
         _reader_ =
-            new BufferedReader(new InputStreamReader(_input_,
+            new CRLFLineReader(new InputStreamReader(_input_,
                                                      __DEFAULT_ENCODING));
         _writer_ =
             new BufferedWriter(new OutputStreamWriter(_output_,
diff --git a/src/main/java/org/apache/commons/net/pop3/POP3.java b/src/main/java/org/apache/commons/net/pop3/POP3.java
index 49b9516dc..ea99adc59 100644
--- a/src/main/java/org/apache/commons/net/pop3/POP3.java
+++ b/src/main/java/org/apache/commons/net/pop3/POP3.java
@@ -29,6 +29,7 @@
 import org.apache.commons.net.MalformedServerReplyException;
 import org.apache.commons.net.ProtocolCommandSupport;
 import org.apache.commons.net.SocketClient;
+import org.apache.commons.net.io.CRLFLineReader;
 
 /***
  * The POP3 class is not meant to be used by itself and is provided
@@ -142,7 +143,7 @@ protected void _connectAction_() throws IOException
     {
         super._connectAction_();
         _reader =
-          new BufferedReader(new InputStreamReader(_input_,
+          new CRLFLineReader(new InputStreamReader(_input_,
                                                    __DEFAULT_ENCODING));
         __writer =
           new BufferedWriter(new OutputStreamWriter(_output_,
diff --git a/src/main/java/org/apache/commons/net/smtp/SMTP.java b/src/main/java/org/apache/commons/net/smtp/SMTP.java
index 8a86212e9..af2c6ec2c 100644
--- a/src/main/java/org/apache/commons/net/smtp/SMTP.java
+++ b/src/main/java/org/apache/commons/net/smtp/SMTP.java
@@ -27,6 +27,7 @@
 import org.apache.commons.net.MalformedServerReplyException;
 import org.apache.commons.net.ProtocolCommandSupport;
 import org.apache.commons.net.SocketClient;
+import org.apache.commons.net.io.CRLFLineReader;
 
 /***
  * SMTP provides the basic the functionality necessary to implement your
@@ -250,7 +251,7 @@ protected void _connectAction_() throws IOException
     {
         super._connectAction_();
         _reader =
-            new BufferedReader(new InputStreamReader(_input_,
+            new CRLFLineReader(new InputStreamReader(_input_,
                                                     encoding));
         _writer =
             new BufferedWriter(new OutputStreamWriter(_output_,
